tutorials/io/binary_serialization_api.rst

```diff
@@ -112,498 +171,722 @@ length of "Float" fields within data structures should be 8, and the offset
 should be ``(offset - 4) * 2 + 4``. The "float" type itself always uses double
 precision.
 
-0: null
+.. _nil-section:
+
+0: NIL
 ~~~~~~~
 
+null
+
 1: :ref:`bool<class_bool>`
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-+----------+-------+-----------+---------------------------+
-| Offset   | Len   | Type      | Description               |
-+==========+=======+===========+===========================+
-| 4        | 4     | Integer   | 0 for False, 1 for True   |
-+----------+-------+-----------+---------------------------+
++----------+----------+-----------+---------------------------+
+| Offset   | Length   | Type      | Description               |
++==========+==========+===========+===========================+
+| 4        | 4        | Integer   | 0 for False, 1 for True   |
++----------+----------+-----------+---------------------------+
 
 2: :ref:`int<class_int>`
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
-If no flags are set (flags == 0), the integer is sent as a 32 bit integer:
-
-+----------+-------+-----------+--------------------------+
-| Offset   | Len   | Type      | Description              |
-+==========+=======+===========+==========================+
-| 4        | 4     | Integer   | 32-bit signed integer    |
-+----------+-------+-----------+--------------------------+
++----------+----------+-----------+-------------------------+
+| Offset   | Length   | Type      | Description             |
++==========+==========+===========+=========================+
+| 4        | 4        | Integer   | 32-bit signed integer   |
++----------+----------+-----------+-------------------------+
```

Comment:
These changes are invalid, my suggestion was to not add the complex different versions between the 32 and 64 bit versions, but this one should be either 64 bits only or the 32 bit as well, but I'd just keep the original text that clarifies how it works